python/ray/rllib/common.py

```diff
@@ -21,13 +21,37 @@
 
 
 class RLLibEncoder(json.JSONEncoder):
+
+    def __init__(self, nan_str="null", **kwargs):
+        super(RLLibEncoder, self).__init__(**kwargs)
+        self.nan_str = nan_str
+
+    def iterencode(self, o, _one_shot=False):
+        if self.ensure_ascii:
+            _encoder = json.encoder.encode_basestring_ascii
+        else:
+            _encoder = json.encoder.encode_basestring
+
+        def floatstr(o, allow_nan=self.allow_nan, nan_str=self.nan_str):
+            if o != o:
```

Comment:
`o` is certainly equal to `o`; perhaps name mismatch? Also, `allow_nan` is not used.

Also, this logic can probably be readably condensed into something like 1 or 2 lines.

Reply:
o != o if o is NaN, so it is correct actually ;)

will try to condense